zhehuaichen/NeMo
# Clean up implementation for SALM paper and sync to NEMO v1.20.0

# Context
We merged Steve’s and Zhehuai’s initial versions of integrating a frozen FC and a frozen LLM PR in early August under https://github.com/zhehuaichen/NeMo/tree/speechllm  mainly in nemo/collections/multimodal. After that, we further developed and obtained the first set of results for SALM paper. 

# What does this PR do ?
1. clean up above extra implementation since the last merge
2. sync to NEMO [v1.20.0](https://github.com/NVIDIA/NeMo/releases/tag/v1.20.0) (to improve readability, do it in https://github.com/zhehuaichen/NeMo/pull/19)

# Testing
1. make sure test_speechllm_models.py passed
2. make sure no degradation on LS

# Pending
Further sync to NEMO v1.21.0 which includes pytorch lightning 2.0 supports
